src/agents/agents.controller.ts

```diff
@@ -44,4 +50,48 @@ export class AgentsController implements OnModuleInit {
   async getAgentStatus() {
     return await this.agentService.getAgentStatus(this.agent);
   }
+
+  @Post('upload_large_file')
+  @UseGuards(
+    new FileTypeGuard([
+      'application/json',
+      'application/zip',
+      'image/jpeg',
+      'image/png',
+    ])
+  )
+  async uploadFile(@Req() req: FastifyRequest) {
+    const logger = new Logger('Upload service');
+    logger.debug({ message: 'The file has been uploaded' });
+    return { status: 'success', data: 'The file has been uploaded.' };
+  }
+
+  @Post('delete_large_file')
+  async deleteUploadFile(@Body() filename: { filename: string }) {
+    const logger = new Logger('Upload service');
+    const filePath = `./uploads/${filename.filename}`;
```

Comment:
This file path handling could be vulnerable to path traversal attacks. 
We should:
- Sanitize filenames using `sanitize-filename`
- Generate unique IDs with `nanoid`
- Add path validation
- Implement strict permissions

Consider creating a dedicated FileSecurityService to handle these concerns.